docker/lib/dependabot/docker/update_checker.rb

```diff
@@ -224,15 +224,17 @@ def tags_from_registry
         @tags_from_registry ||=
           begin
             client = docker_registry_client
-
-            client.tags(docker_repo_name, auto_paginate: true).fetch("tags")
+            repo = docker_repo_name
+            client.tags(repo, auto_paginate: true).fetch("tags")
           rescue *transient_docker_errors
             attempt ||= 1
             attempt += 1
             raise if attempt > 3
 
             retry
           end
+      rescue DockerRegistry2::NotFound
+        raise DockerRegistry2::NotFound, "404 Not Found. Image https://#{registry_hostname}/#{repo} not found"
```

Comment:
Would it make sense to improve this upstream as well, so we can eventually remove this? https://github.com/deitch/docker_registry2/blob/970b4aafd1259e4867293f9a6be180a46730b366/lib/registry/registry.rb#L284-L285. Just thinking that the enhancement may be useful for all users of the `docker_registry2` gem.

Reply:
I agree with you. I will create an issue to improve this upstream. Thank you for pointing it out.

Reply:
Issue created in upstream: https://github.com/deitch/docker_registry2/issues/67